src/script/src/python/pyo3/vector_impl.rs

```diff
@@ -299,6 +299,49 @@ impl PyVector {
             })?;
             let ret = Self::from(ret).into_py(py);
             Ok(ret)
+        } else if let Ok(slice) = needle.downcast::<PySlice>(py) {
+            let indices = slice.indices(self.len() as i64)?;
+            let (start, stop, step, _slicelength) = (
+                indices.start,
+                indices.stop,
+                indices.step,
+                indices.slicelength,
+            );
+            if start < 0 {
+                return Err(PyValueError::new_err(format!(
+                    "Negative start is not supported, found {start} in {indices:?}"
+                )));
+            } // Negative stop is supported, means from "indices.start" to the actual start of the vector
+            let vector = self.as_vector_ref();
+
+            let mut buf = vector
+                .data_type()
+                .create_mutable_vector(indices.slicelength as usize);
+            let v = if indices.slicelength == 0 {
+                buf.to_vector()
+            } else {
+                if indices.step > 0 {
+                    let range = if stop == -1 {
```

Comment:
Can we calculate `stop` from `slicelength`? It seems `|stop - start| = slicelength`?

Reply:
I originally thought Python's slicing length is the final results of slicing, like floor((stop-start)/step), gonna check if this is correct

Reply:
> Can we calculate `stop` from `slicelength`? It seems `|stop - start| = slicelength`?

https://docs.python.org/3/c-api/slice.html#c.PySlice_GetIndicesEx
it seems slicelength store length of slice, so unless step==1 it's not equal?